pkg/controller/istio/istio_controller.go

```diff
@@ -477,14 +480,19 @@ func (r *ReconcileIstio) getMeshNetworks(config *istiov1beta1.Istio, logger logr
 			}
 		}
 
-		meshNetworks[remoteIstio.Name] = istiov1beta1.MeshNetwork{
-			Endpoints: []istiov1beta1.MeshNetworkEndpoint{
-				{
-					FromRegistry: remoteIstio.Name,
-				},
-			},
-			Gateways: gateways,
+		networkName := remoteIstio.Spec.NetworkName
+		// to be backward compatible, the network name is the CR name if not specified
+		if networkName == "" {
+			networkName = remoteIstio.Name
+		}
+
+		if meshNetworks[networkName] == nil {
+			meshNetworks[networkName] = &istiov1beta1.MeshNetwork{}
 		}
+		meshNetworks[networkName].Endpoints = append(meshNetworks[networkName].Endpoints, istiov1beta1.MeshNetworkEndpoint{
+			FromRegistry: remoteIstio.Name,
```

Comment:
Might be missing something here, but shouldn't this be:
```suggestion
			FromRegistry: networkName,
```

Reply:
The registry is the name of the cluster, not the network.

Reply:
got it, thanks